frontend/src/pages/patients/PatientsPage.tsx

```diff
@@ -65,9 +71,101 @@ function patientAliasFilterWhereVariables(value: string): PatientAliasWhere[] {
   }
 }
 
-export const PatientsPage: React.FunctionComponent = (props) => {
+export default function PatientsPage({ setUserEmail }: { setUserEmail: any }) {
   const params = useParams();
 
+  const [searchWithMRNs, setSearchWithMRNs] = useState(false);
+  const [patientIdsTriplets, setPatientIdsTriplets] = useState<
+    PatientIdsTriplet[]
+  >([]);
+  const [patientsListColumns, setPatientsListColumns] =
+    useState(PatientsListColumns);
+  const [popupOpen, setPopupOpen] = useState(false);
+
+  useEffect(() => {
+    function handleLogin(event: any) {
+      if (event.origin !== "http://localhost:4001") return;
+      setUserEmail(event.data);
+      setPopupOpen(false);
+    }
+
+    if (popupOpen) {
+      window.addEventListener("message", handleLogin);
+      return () => {
+        window.removeEventListener("message", handleLogin);
+      };
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [popupOpen]);
```

Comment:
See my comment [here](https://github.com/mskcc/smile-dashboard/pull/101/files#r1367406307) for more info.